### PR TITLE
[Better Tablet Products] Add margins for details view. Navigation state preserving

### LINKED ISSUE #10847: [Better Tablet Products] Add Margins for details view

### LINKED ISSUE #10873: [Better Tablet Products] Fix image selection on the 2 pane layout

### LINKED ISSUE #10944: [Better Tablets Products] Save navigation progress on activity restart and config change

### LINKED ISSUE #10945: [Better Tablets Products] Keep navigation progress when coming from 2 pane layout to 1 pane (e.g. split screen)

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #10847
Closes: #10945
Closes: #10944 
Closes: #10873

### Description

The PR: 
* Adds margins to the fragments that are "details" (keep in mind that it also adds margin to the toolbar of the main details screen, and that will be fixed later)
* Let the details navigation controller keep its progress by not recreating it
* When the screen size is too small for the tablet, it transitions to the one pane view, keeping the progress (navigation "up" is not working yet on the root details fragment and will be fixed later)

### Images/gif

* Try the changes mentioned above (check the video for better understanding)
* Make sure no regressions introduced when the FF is OFF

https://github.com/woocommerce/woocommerce-android/assets/4923871/f1f77aee-4a4f-4327-885f-ff7999d16c5c

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/util/TabletLayoutSetupHelper.kt [hunk 1]
```diff
@@ -18,83 +23,177 @@ class TabletLayoutSetupHelper @Inject constructor(
...
-    private lateinit var navHostFragment: NavHostFragment
-
-    fun onViewCreated(screen: Screen) {
-        this.screen = screen
-        screen.lifecycleKeeper.addObserver(this)
-
-        if (isTabletLogicNeeded()) {
-            initNavFragment(screen.secondPaneNavigation)
-            adjustUIForScreenSize(screen.twoPaneLayoutGuideline)
+            screen.listFragment.childFragmentManager.registerFragmentLifecycleCallbacks(
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/util/TabletLayoutSetupHelper.kt:56]: @samiuelson we may want to reuse this for order's margins as well, to keep it consistent

Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/util/TabletLayoutSetupHelper.kt:56]: cc @jd-alexander

**Review by backwardstruck — APPROVED** [review]
Tested with FF on and off. Code changes make sense. I didn't find any regressions with the FF off.

**Pull request merged**